pkg/policy/resolve.go

```diff
@@ -86,6 +102,85 @@ func getSecurityIdentities(labelsMap cache.IdentityCache, selector *api.Endpoint
 	return identities
 }
 
+// DistillPolicy filters down the specified IdentityPolicy (which acts upon
+// selectors) into a set of concrete map entries based on the specified
+// identityCache. These can subsequently be plumbed into the datapath.
+//
+// Must be performed while holding the Repository lock.
+func (p *IdentityPolicy) DistillPolicy(policyOwner PolicyOwner, identityCache cache.IdentityCache) *EndpointPolicy {
+
+	calculatedPolicy := &EndpointPolicy{
+		IdentityPolicy:          p,
+		PolicyMapState:          make(MapState),
+		PolicyOwner:             policyOwner,
+		DeniedIngressIdentities: cache.IdentityCache{},
+		DeniedEgressIdentities:  cache.IdentityCache{},
+	}
+
+	labels := policyOwner.GetSecurityIdentity().LabelArray
+	ingressCtx := SearchContext{
+		To:                            labels,
+		rulesSelect:                   true,
+		skipL4RequirementsAggregation: true,
+	}
+
+	egressCtx := SearchContext{
+		From:                          labels,
+		rulesSelect:                   true,
+		skipL4RequirementsAggregation: true,
+	}
+
+	if option.Config.TracingEnabled() {
+		ingressCtx.Trace = TRACE_ENABLED
+		egressCtx.Trace = TRACE_ENABLED
+	}
+
+	if p.IngressPolicyEnabled {
+		for identity, labels := range identityCache {
+			ingressCtx.From = labels
+			egressCtx.To = labels
```

Comment:
I think this line, while harmless, can be removed - the ingressContext is the only thing which is being evaluated here. This is in the old code as well.

Reply:
Yeah, I kept it the same just so that the individual commits are direct refactors. I can look into this.